services/nginx/conf.d/shift.conf

```diff
@@ -70,6 +70,16 @@ server {
         alias /opt/legacy/cal;
     }
 
+    # handle images of the standard shift format 
+    # ex. /eventimages/9248-124.png
+    #
+    location ~ /eventimages/(\d+)(?:-\d+).(\w+) {
```

Comment:
What does `~` do in this context? Not familiar with this conf file syntax, just noticed that the existing `eventimages` line below doesn't have it.

Reply:
the `~` does [nginx regex matching](https://www.digitalocean.com/community/tutorials/nginx-location-directive). 

it pairs with `alias /opt/backend/eventimages/$1.$2;` ( on line 77  ) to send a request like `/eventimages/9248-124.png` to `/opt/backend/eventimages/9248.png` -- stripping off the second number ( ex. `124` ). the second number is the cache busting "version" number. the real file on disk doesn't have that second number.

if it doesn't match that regex, nginx will fall through to the next line and request an image without trying to rewrite its name. ( in case someone has an old urls which lacks the dash. )